pyproject.toml

```diff
@@ -3,6 +3,7 @@ requires = [
     "wheel",
     "setuptools",
     "Cython>=0.29.18",
+    "cysignals",
```

Comment:
As you've alluded to in the PR description, addition of a new dependency to SciPy would indeed require some discussion.

@pv and @rgommers probably for initial thoughts on adding a dependency to improve ctrl+C handling in Cython.

I'm probably a little on the "against" side if we could have something that isn't too ugly internally instead, but let's let others chime in.

Reply:
I should have made it clearer that this is nothing more than an RFC in its current state. The sig_on sig_off indeed looks ugly.